collect_app/src/test/java/org/odk/collect/android/database/DatabaseMediaFileRepositoryTest.java

```diff
@@ -15,6 +18,32 @@ public void whenThereIsNoFormForgivenFormIdAndVersion_shouldGetAllMethodReturnEm
         FormsDao formsDao = mock(FormsDao.class);
         when(formsDao.getFormMediaPath("1", "1")).thenReturn(null);
 
-        assertThat(new DatabaseMediaFileRepository(formsDao).getAll("1", "1").size(), is(0));
+        assertThat(new DatabaseMediaFileRepository(formsDao, new FileUtil()).getAll("1", "1").size(), is(0));
+    }
+
+    @Test
+    public void whenMediaFolderFileDoesNotExist_shouldGetAllMethodReturnEmptyArray() {
+        FormsDao formsDao = mock(FormsDao.class);
+        FileUtil fileUtil = mock(FileUtil.class);
```

Comment:
But then we would need to move this test to android tests you you want to check if a file really exists or not. Would that improve anything?

Reply:
It should work in normal `test` as we're just using `File` rather than any Android specific code right? I think if you just have `FormsDao#getFormMediaPath` return `"/samplePath/"` (like you're doing) then the as that path doesn't exist you should end up creating the environment you need to check that `getAll` returns an empty list.